tests/unit/log_buf_test.cc

```diff
@@ -35,16 +35,13 @@ SEASTAR_TEST_CASE(log_buf_realloc) {
 
     auto it = b.back_insert_begin();
 
-    BOOST_REQUIRE_EQUAL(reinterpret_cast<uintptr_t>(&*it), external_buf_ptr);
```

Comment:
Why this has to be removed?

Reply:
Because after my change `&*it` is the address of the iterator itself, not the underlying buffer.